pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/IngestionDelayTracker.java

```diff
@@ -174,6 +188,22 @@ private long getIngestionDelayMs(long ingestionTimeMs) {
     return agedIngestionDelayMs;
   }
 
+  private long getPartitionOffsetLag(IngestionOffsets offset) {
+    if (offset == null) {
+      return 0;
+    }
+    StreamPartitionMsgOffset msgOffset = offset._offset;
+    StreamPartitionMsgOffset latestOffset = offset._latestOffset;
+
+    // Compute aged delay for current partition
+    // TODO: Support other types of offsets
+    if (!(msgOffset instanceof LongMsgOffset && latestOffset instanceof LongMsgOffset)) {
+      return 0;
```

Comment:
What are other types that can fall here? Basically we need to be able to explain this metric.

Reply:
Other than this, Kinesis returns a BigInteger offset and Pulsar returns a string offset that is composed of 3 longs